deps/v8/src/builtins/builtins-promise-gen.cc

```diff
@@ -509,8 +509,8 @@ Node* PromiseBuiltinsAssembler::InternalPerformPromiseThen(
     BIND(&if_existingcallbacks);
     {
       Label if_singlecallback(this), if_multiplecallbacks(this);
-      BranchIfJSObject(existing_deferred_promise, &if_singlecallback,
-                       &if_multiplecallbacks);
+      Branch(HasInstanceType(existing_deferred_promise, FIXED_ARRAY_TYPE),
```

Comment:
You could also use `Branch(IsFixedArray(existing_deferred_promise), ...)` here.

Reply:
/cc @sigurdschneider